ocaml/bytecomp/bytegen.ml

```diff
@@ -504,30 +504,30 @@ let comp_primitive stack_info p sz args =
   (* In bytecode, nothing is ever actually stack-allocated, so we ignore the
      array modes (allocation for [Parrayref{s,u}], modification for
      [Parrayset{s,u}]). *)
-  | Parrayrefs (Pgenarray_ref _) -> Kccall("caml_array_get", 2)
-  | Parrayrefs (Pfloatarray_ref _) -> Kccall("caml_floatarray_get", 2)
-  | Parrayrefs (Paddrarray_ref | Pintarray_ref) ->
+  | Parrayrefs (Pgenarray_ref _, _index_kind) -> Kccall("caml_array_get", 2)
```

Comment:
Shouldn't these at least assert that `index_kind` is a tagged int ? If it's not the case, I'd expect the bytecode interpreter to crash/misinterpret the instruction.

Reply:
Yeah, the PR is still very much work-in-progress. This needs to be fixed

Reply:
Added some new runtime functions for this to work properly.